src/layouts/Folders.jsx

```diff
@@ -0,0 +1,132 @@
+import FolderContent from '../components/folders/FolderContent';
+
+import {
+    parseDirectoryFile,
+    convertFolderOrderToArray,
+    retrieveSubfolderContents,
+} from '../utils'
+
+// Import API
+import { getDirectoryFile } from '../api';
+
+// Import styles
+import elementStyles from '../styles/isomer-cms/Elements.module.scss';
+import contentStyles from '../styles/isomer-cms/pages/Content.module.scss';
+
+// Constants
+const FOLDER_CONTENTS_KEY = 'folder-contents'
+
+const Folders = ({ match, location }) => {
+    const { siteName, folderName, subfolderName } = match.params;
+
+    const { data: folderContents, error } = useQuery(
+        FOLDER_CONTENTS_KEY,
+        () => getDirectoryFile(siteName, folderName)
+    );
+    const [isRearrangeActive, setIsRearrangeActive] = useState(false)
+    const [directoryFileSha, setDirectoryFileSha] = useState('')
+    const [folderOrder, setFolderOrder] = useState([])
+
+    useEffect(() => {
+        if (folderContents && folderContents.data) {
+            const parsedFolderContents = parseDirectoryFile(folderContents.data.content)
+            setDirectoryFileSha(folderContents.data.sha)
+
+            if (subfolderName) {
+                setFolderOrder(retrieveSubfolderContents(parsedFolderContents, subfolderName))
+            } else {
+                setFolderOrder(convertFolderOrderToArray(parsedFolderContents))
+            }
+        }
+    }, [folderContents, subfolderName])
+
+    const toggleRearrange = () => { setIsRearrangeActive((prevState) => !prevState) }
+
+    return (
+        <>
+          <Header
+            backButtonText={`Back to ${subfolderName ? folderName : 'Workspace'}`}
+            backButtonUrl={`/sites/${siteName}/${subfolderName ? `folder/${folderName}` : 'workspace'}`}
+          />
+          {/* main bottom section */}
+          <div className={elementStyles.wrapper}>
+            <Sidebar siteName={siteName} currPath={location.pathname} />
+            {/* main section starts here */}
+            <div className={contentStyles.mainSection}>
+              {/* Page title */}
+              <div className={contentStyles.sectionHeader}>
+                <h1 className={contentStyles.sectionTitle}>{folderName}</h1>
+              </div>
+              {/* Info segment */}
+              <div className={contentStyles.segment}>
+                <i className="bx bx-sm bx-bulb text-dark" />
+                <span><strong className="ml-1">Pro tip:</strong> You can make a new section by creating sub folders</span>
+              </div>
+              {/* Segment divider  */}
+              <div className={contentStyles.segmentDividerContainer}>
+                <hr className="w-100 mt-3 mb-5" />
+              </div>
+              {/* Collections title */}
+              <div className={contentStyles.segment}>
+                <span>
+                    My workspace >
+                    {
+                        folderName && !subfolderName
```

Comment:
design suggestion - would it be more intuitive to users if this was a link that allowed users to navigate between folders, rather than having to look in the top nav bar?

Reply:
I like that idea - implemented in [
5cd1bdd](https://github.com/isomerpages/isomercms-frontend/pull/350/commits/5cd1bdd8a3dbb7f5f3d62e5ac2e73c2644c8f624)